docs/src/main/asciidoc/datasource.adoc

```diff
@@ -436,69 +381,94 @@ AgroalDataSource usersDataSource;
-at least one build time property so Quarkus knows to create the datasource. You will usually either specify
-the `db-kind` property, or explicitly enable Dev Services: `quarkus.datasource."name".devservices.enabled=true`.
+==== Named datasources
+
+When using Dev Services, the default datasource will always be created, but to specify a named datasource, you need to have at least one build time property so Quarkus can detect how to create the datasource.
+
+You will usually specify the `db-kind` property or explicitly enable Dev Services by setting `quarkus.datasource."name".devservices.enabled=true`.
 
 [[in-memory-databases]]
-== Testing with in-memory databases
+=== Testing with in-memory databases
+
+Some databases like H2 and Derby are commonly used in the _embedded mode_ as a facility to run integration tests quickly.
 
-Some databases like H2 and Derby are commonly used in "embedded mode" as a facility to run quick integration tests.
+It is recommended to use the real database you intend to use in production, but it’s possible to use JVM-powered databases for the scenarios when the ability to run quick integration tests is required.
 
-Our suggestion is to use the real database you intend to use in production; container technologies made this simple enough so you no longer have an excuse. Still, there are sometimes
-good reasons to also want the ability to run quick integration tests using the JVM powered databases,
-so this is possible as well.
+NOTE: While configuring Derby to use the embedded engine works as usual in JVM mode, such an application will not compile into a native executable. It is because Quarkus Derby extensions do not support embedding the entire database engine into a native executable, and it only covers making the JDBC client code compatible with the native compilation step.
 
-It is important to remember that when configuring Derby to use the embedded engine,
-this will work as usual in JVM mode but such an application will not compile into a native executable, as the Quarkus extensions only cover for making the JDBC client code compatible with the native compilation step: embedding the whole database engine into a native executable is currently not implemented.
+==== Support and limitations
 
-Compiling H2 into a native-image is now possible (it didn't use to be) but this is considered experimental; even if this is technically possible,
-we would still recommend to consider carefully why you would want to embed H2 within your native-image.
+Embedded databases (H2 and Derby) work in JVM mode.
+For native mode, the following limitations apply:
 
-If you plan to run such integration tests in the JVM exclusively, it will of course work as usual.
+* Derby cannot be embedded into the application in native mode, and only remote connection is supported.
+[IMPORTANT]
+====
+Embedding H2 within your native image is not recommended. 
```

Comment:
Is there any more detail on why this is not recommended? See https://github.com/quarkusio/quarkus/issues/6320#issuecomment-2157743619

Reply:
As far as I know, just because of uncertainty regarding bugs, feature coverage, performance. It started out as something experimental and didn't change much since then.

We haven't had many bug reports either, so you can interpret that any way you like: either nobody uses H2 in native images, or it works fine.